pkgs/tools/networking/slack-cli/default.nix

```diff
@@ -0,0 +1,47 @@
+# slack-cli must be configured using the SLACK_CLI_TOKEN environment variable.
+# Using `slack init` will not work because it tries to write to the Nix store.
+#
+# There is no reason that we couldn't change the file path that slack-cli uses
+# for token storage, except that it would make the Nix package inconsistent with
+# upstream and other distributions.
+
+{ stdenv, lib, writeShellScriptBin, fetchFromGitHub, curl, jq }:
+
+let
+  wrapper = writeShellScriptBin "slack" ''
+    [ "$1" = "init" -a -z "$SLACK_CLI_TOKEN" ] && cat >&2 <<-'MESSAGE'
+    WARNING: slack-cli must be configured using the SLACK_CLI_TOKEN environment
+    variable. Using `slack init` will not work because it tries to write to the
+    Nix store.
+
+    MESSAGE
+
+    export PATH=${lib.makeBinPath [ curl jq ]}:"$PATH"
+    exec "$(dirname "$0")/.slack-wrapped" "$@"
+  '';
```

Comment:
You can use

```nix
let
  wrapper = writeShellScriptBin "slack" ''
    ...
    exec "${slack-cli}/bin/.slack-wrapped"
  '';

  slack-cli = stdenv.mkDerivation rec {
    ...
  };

in slack-cli
```

(or similar) to not have to use this dirname hack

Reply:
I tried this! (Or something very similar.) It was infinitely recursive though. Was I doing something wrong? It makes sense to me I think – `wrapper` takes `slack-cli`'s output path as an input, and `slack-cli`'s output path depends on all of its inputs, one of which is `wrapper`.

Reply:
@Infinisil that does work (just pushed), although it puts me in shell-escape hell…

Reply:
Yeah that is a bit of a disadvantage..

Reply:
What do you think then? I think that, having seen it, I actually prefer this version. Even if it was a bit annoying to write, it’s less hacky.

Reply:
Agreed!